packages/react/src/hooks/useGridLayout.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  GRID_LAYOUTS,
+  selectGridLayout,
+  TrackBundleWithPlaceholder,
+} from '@livekit/components-core';
+import { GridLayout } from '@livekit/components-core/dist/helper/grid-layouts';
+import * as React from 'react';
+import { useSize } from '../helper';
+import { useVisualStableUpdate } from './useVisualStableUpdate';
+
+/**
+ * The useGridLayout hook tries to select the best layout to fit all tiles.
+ * If the available screen space is not enough, it will reduce the number of maximum visible
+ * tiles and select a layout that still works visually within the given limitations.
+ * As the order of tiles changes over time, the hook tries to keep visual updates to a minimum
+ * while trying to display important tiles such as speaking participants or screen shares.
+ */
+export function useGridLayout(
+  /** HTML element that contains the gird. */
+  containerElement: React.RefObject<HTMLDivElement>,
+  /** Direct child of the `containerElement`. */
+  gridElement: React.RefObject<HTMLDivElement>,
```

Comment:
could we automatically detect that `gridElement` instead of having to pass both explicitly?

Reply:
Good idea. I wonder if we can get by with just one element? Or would it be a problem if the grid is also applied to the containing element?

Reply:
not sure, but worth a try!